enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/RestAPI.java

```diff
@@ -344,6 +344,16 @@ private void loadCustomFilters(APIConfig apiConfig) {
     }
 
     private void populateRemoveAndProtectedHeaders(RequestContext requestContext) {
+        // If the resource has disabled security, then the authorization headers are passed as it is.
+        // Expectation is that the backend should validate the authorization header if it is not processed
+        // at the gateway level.
+        // It is required to check if the resource Path is not null, because when CORS preflight request handling, or
+        // generic OPTIONS method call happens, matchedResourcePath becomes null.
+        if (requestContext.getMatchedResourcePaths() != null && requestContext.getMatchedResourcePaths().size() > 0 &&
```

Comment:
with graphql changes in place it seems there could be an ambiguity on what would happen

Reply:
my bad. It won't happen since there is a separate api type for graphql

Reply:
This is RestAPI.java. I think we have to do the same changes for GraphQL.java.